### PR TITLE
Added npm-package template

### PR DESCRIPTION
**Pull request by mrmihi**
- Added _npm-package_ template from the [sliit-foss/npm-catalogue](https://github.com/sliit-foss/npm-catalogue)
- The list of cleaned files and directories from the original repo

root
└── packages
.gitmodules
LICENSE

### CODE REVIEW

File: npm-package/turbo.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    "release": {
+      "dependsOn": [],
+      "outputs": []
+    },
+    "test": {
+      "dependsOn": ["build"],
+      "outputs": ["coverage/**"]
+    }
+  },
+  "globalEnv": ["DISABLE_FUNCTION_TRACING"]
```
Comment (Akalanka47000) [inline npm-package/turbo.json]: Let's remove this key from the globalEnv, it's specific to the @sliit-foss/functions package

File: npm-package/.github/actions/unit-test/action.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        node-version: '16.x'
+        registry-url: 'https://registry.npmjs.org'
+
+    - name: Install dependencies
+      run: npm install -g pnpm@7.5.0 && pnpm install
+      shell: bash
+
+    - name: Run unit tests
+      run: pnpm --filter ${{ inputs.package }} run test
+      shell: bash
```
Comment (Akalanka47000) [inline npm-package/.github/actions/unit-test/action.yml]: Let's remove this file, it's not being used

File: npm-package/.github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+      - name: Install dependencies
+        run: npm install -g pnpm@8 && pnpm install --ignore-scripts
+      
+      - name: Run unit tests
+        shell: bash
+        run: pnpm test
+        env:
+          GITHUB_ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN_GITHUB }}
+          FIREBASE_CONFIG: ${{ secrets.FIREBASE_CONFIG }}
```
Comment (Akalanka47000) [inline npm-package/.github/workflows/tests.yml]: These env variables won't be needed as well, they were for specific tests

File: npm-package/package.json [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    "format": "prettier --write \"**/*.{js,ts,tsx,md}\"",
+    "prepare": "husky install",
+    "prerelease": "turbo run prerelease",
+    "release": "turbo run release --concurrency=1",
+    "test": "dotenv -- turbo run test"
+  },
+  "dependencies": {
+    "@babel/core": "7.20.12",
+    "@babel/preset-env": "7.20.2",
+    "@sliit-foss/automatic-versioning": "workspace:*",
```
Comment (Akalanka47000) [inline npm-package/package.json]: Let's point this to the published package in the npm registry, this installation won't work as it's no longer in this workspace

File: npm-package/package.json [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    "jest": "29.4.1",
+    "prettier": "2.8.3",
+    "rimraf": "4.1.2",
+    "shx": "0.3.4",
+    "turbo": "1.7.3"
+  },
+  "devDependencies": {
+    "@commitlint/cli": "17.4.2",
+    "@commitlint/config-conventional": "17.4.2",
+    "@sliit-foss/eslint-config-internal": "workspace:*",
```
Comment (Akalanka47000) [inline npm-package/package.json]: The same applies here as for the @sliit-foss/automatic-versioning package, let's point this to the NPM registry

File: npm-package/package.json [hunk 6]
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "npm-catalogue",
+  "version": "1.0.0",
+  "private": true,
+  "workspaces": [
+    "packages/*",
+    "plugins/*"
```
Comment (Akalanka47000) [inline npm-package/package.json]: Let's remove this `plugins` workspace, both from here and the [pnpm-workspace.yaml](https://github.com/sliit-foss/nodejs-templates/pull/1/files#diff-5f647fbcca6b22bfe7ced86593890e1f26e092b32b76cf2c811457748f959ad1) file. Let's remove the babel plugin and the eslint plugins included as well and add a simple package like one which contains a few calculator functions to the packages workspace to keep things simple

**Review by Akalanka47000 — APPROVED** [review]

File: npm-package/turbo.json (hunk 1 above)
Comment (mrmihi) [reply npm-package/turbo.json]: Will we have to remove both the key and value or just the value ["DISABLE_FUNCTION_TRACING"] ?

File: npm-package/package.json (hunk 6 above)
Comment (mrmihi) [reply npm-package/package.json]: How about "lodash"?

File: npm-package/turbo.json (hunk 1 above)
Comment (Akalanka47000) [reply npm-package/turbo.json]: Let's remove just the value

File: npm-package/package.json (hunk 6 above)
Comment (Akalanka47000) [reply npm-package/package.json]: Didn't quite get this, do you mean a dummy package named lodash ?

Comment (mrmihi) [reply npm-package/package.json]: Added an example package with a simple calc function.

**Review by Akalanka47000 — APPROVED** [review]

**Pull request merged**